test/integration/apiserver/apiserver_test.go

```diff
@@ -618,11 +615,9 @@ func TestListResourceVersion0(t *testing.T) {
 
 	for _, tc := range testcases {
 		t.Run(tc.name, func(t *testing.T) {
-			_, ctx := ktesting.NewTestContext(t)
-			ctx, cancel := context.WithCancel(ctx)
-			defer cancel()
+			tCtx := ktesting.Init(t)
 
-			clientSet, _, tearDownFn := framework.StartTestServer(ctx, t, framework.TestServerSetup{
+			clientSet, _, tearDownFn := framework.StartTestServer(tCtx, t, framework.TestServerSetup{
```

Comment:
The next step would be to change `framework.StartTestServer` so that the caller doesn't need to set up a context and a `ktesting.KContext` is returned, replacing (in this case) `clientSet` with `kCtx` and replacing `tCtx` with `kCtx`.

Reply:
This could be done in a single step, but it then becomes larger because all users of `framework.TestServerSetup` need to be touched.